ground/src/main/java/com/google/android/ground/persistence/remote/firestore/schema/LoiConverter.kt

```diff
@@ -37,52 +34,48 @@ object LoiConverter {
     const val POLYGON_TYPE = "Polygon"
     const val GEOMETRY_COORDINATES = "coordinates"
     const val GEOMETRY = "geometry"
+    private val geometryFactory = GeometryFactory()
 
     @JvmStatic
     @Throws(DataStoreException::class, NotImplementedError::class)
     fun toLoi(survey: Survey, doc: DocumentSnapshot): LocationOfInterest {
         val loiId = doc.id
-        val loiDoc = checkNotNull(doc.toObject(LoiDocument::class.java), "LOI data")
-        val geometryMap = checkNotNull(loiDoc.geometry, "geometry")
+        val loiDoc =
+            DataStoreException.checkNotNull(doc.toObject(LoiDocument::class.java), "LOI data")
+        val geometryMap = DataStoreException.checkNotNull(loiDoc.geometry, "geometry")
         // TODO: Return `Result` instead of throwing exception.
         val geometry = GeometryConverter.fromFirestoreMap(geometryMap).getOrThrow()
 
         // As an interim solution, we map geometries to existing LOI types.
         // TODO: Get rid of LOI subclasses and just use Geometry on LOI class.
         when (geometry.geometryType) {
-            "Point" -> {
-                val builder = PointOfInterest.newBuilder()
-                builder.setPoint(
-                    Point.newBuilder().setLatitude(geometry.coordinate.x)
-                        .setLongitude(geometry.coordinate.y).build()
+            "Point" ->
```

Comment:
One more reason to create our own model classes (runtime String bindings are more error prone as opposed to static enums)